tests/e2e/test_query_assets_download_media.py

```diff
@@ -0,0 +1,488 @@
+    """test download_assets on text project"""
+    with TemporaryDirectory() as tmp_dir:
+        assets = kili.assets(
+            project_id=src_project_text["id"],
+            fields=["externalId", "content", "jsonContent", "project.inputType"],
+        )
+
+        media_dl = MediaDownloader(
+            tmp_dir,
+            src_project_text["id"],
+            jsoncontent_field_added=False,
+            project_input_type="TEXT",
+        )
+        assets = media_dl.download_assets(assets)
+
+        assert_helper_for_asset(
+            assets[0],
+            expected_externalid="text1",
+            check_is_file=assets[0]["content"],
+            expected_content=os.path.join(str(tmp_dir.resolve()), "text1"),
+            expected_json_content="",
+        )
+        assert_helper_for_asset(
+            assets[1],
+            expected_externalid="text1.txt",
+            check_is_file=assets[1]["content"],
+            expected_content=os.path.join(str(tmp_dir.resolve()), "text1.txt"),
+            expected_json_content="",
+        )
+        # asset[2] is jsoncontent richtext
+        assert_helper_for_asset(
+            assets[2],
+            expected_externalid="richtext",
+            check_is_file=assets[2]["jsonContent"],
+            expected_content="",
+            expected_json_content=os.path.join(str(tmp_dir.resolve()), "richtext"),
+        )
+
+
+@pytest.fixture()
+def src_project_big_image(kili):
+    """create image project with big images and jsonContent"""
+    interface = {
+        "jobs": {
+            "DETECTION": {
+                "mlTask": "OBJECT_DETECTION",
+                "tools": ["rectangle"],
+                "instruction": "Is there a defect ? Where ? What kind ?",
+                "required": 0,
+                "isChild": False,
+                "content": {
+                    "categories": {
+                        "DEFECT_CLASS_1": {"name": "defect of class 1"},
+                        "DEFECT_CLASS_2": {"name": "defect of class 2"},
+                        "DEFECT_CLASS_3": {"name": "defect of class 3"},
+                        "DEFECT_CLASS_4": {"name": "defect of class 4"},
+                    },
+                    "input": "radio",
+                },
+            }
+        }
+    }
+
+    project = kili.create_project(
+        input_type="IMAGE",
+        json_interface=interface,
+        title="test_query_assets_download_media.py",
+        description="test_query_assets_download_media.py",
+    )
+
+    with NamedTemporaryFile(mode="w+b", suffix=".png", delete=False) as temp:
+        width = 3800
+        height = 3800
+
+        img = Image.new(mode="RGB", size=(width, height))
+
+        # fill image with random data to avoid image compression
+        random_data = [random.randint(0, 255) for _ in range(width * height)]
+        img.putdata(list(zip(random_data, random_data, random_data)))  # type: ignore
+
+        img.save(temp, "PNG")
```

Comment:
it could be coded within another PR but we should assert that the size of the created image is bigger than LARGE_IMAGE_THRESHOLD_SIZE to be sure that it is upload asynchronously by the backend

Reply:
ah yes indeed!